File: math/168.py
# Meta, Microsoft, Zenefit's

# Tc = O(log(26)N) [N = is the column number] and space is same for the output string

    def convertToTitle(self, columnNumber: int) -> str:
        result = []
        while columnNumber > 0:
            columnNumber -= 1
            remainder = columnNumber % 26
            ch = chr(ord('A') + remainder)
            result.append(ch)
            columnNumber //= 26
        result.reverse()
        return ''.join(result)

File: math/168.js
/**
 * @param {number} columnNumber
 * @return {string}
 */
var convertToTitle = function(cn) {
    let result = []
    while(cn > 0){
        cn -= 1;
        let remainder = cn % 26
        let ch = String.fromCharCode('A'.charCodeAt(0) + remainder)
        result.push(ch)
        cn = Math.floor(cn / 26)
    }
    result.reverse()
    return result.join('')
};